src/java.base/share/classes/java/security/PEMDecoder.java

```diff
@@ -0,0 +1,486 @@
+     * The {@code InputStream} read pointer will remain at the end of
+     * the PEM footer unless an IOException occurs.
+     *
+     * <p> This method returns a Java API cryptographic object,
+     * such as a {@code PrivateKey}, if the PEM type is supported.
+     * Any non-PEM data preceding the PEM header is ignored by the decoder.
+     * If no cryptographic object is found, a {@link PEMRecord} will be
+     * returned containing the type identifier and Base64-encoded data.
+     * Any non-PEM data preceding the PEM header will be stored in
+     * {@code leadingData}.
+     *
+     * <p> If no PEM data is found, an {@code IllegalArgumentException} is
+     * thrown.
+     *
+     * @param is InputStream containing PEM data
+     * @return a {@code DEREncodable}
+     * @throws IOException on IO error with the {@code InputStream}
+     * @throws EOFException the end of {@code InputStream} has been
+     * unexpectedly reached.
+     * @throws IllegalArgumentException on error in decoding or no PEM data
+     * found
+     * @throws NullPointerException when {@code is} is null.
+     */
+    public DEREncodable decode(InputStream is) throws IOException {
+        Objects.requireNonNull(is);
+        PEMRecord pem = Pem.readPEM(is);
+        return decode(pem);
+    }
+
+    /**
+     * Decodes and returns a {@code DEREncodable} of the specified class from
+     * the given PEM string. {@code tClass} must extend {@link DEREncodable}
+     * and be an appropriate class for the PEM type.
+     *
+     * <p> This method reads the {@code String} until PEM data is found
+     * or the end of the {@code String} is reached.  If no PEM data is found,
+     * an {@code IllegalArgumentException} is thrown.
+     *
+     * <p> If the class parameter is {@code PEMRecord.class},
+     * a {@linkplain PEMRecord} is returned containing the
+     * type identifier and Base64 encoding. Any non-PEM data preceding
+     * the PEM header will be stored in {@code leadingData}.  Other
+     * class parameters will not return preceding non-PEM data.
+     *
+     * <p> Input consumed by this method is read in as
+     * {@link java.nio.charset.StandardCharsets#UTF_8 UTF-8}.
+     *
+     * <p> For all other class parameters, {@code IllegalArgumentException} is
+     * thrown when not PEM data is found.  All non-PEM data is ignored.
+     *
+     * @param <S> Class type parameter that extends {@code DEREncodable}.
+     * @param str the String containing PEM data
+     * @param tClass the returned object class that implements
+     * {@code DEREncodable}.
+     * @return a {@code DEREncodable} specified by {@code tClass}
+     * @throws IllegalArgumentException on error in decoding or no PEM data
+     * found
+     * @throws ClassCastException if {@code tClass} is invalid for the PEM type.
+     * @throws NullPointerException when any input values are null.
+     *
+     * @see PEMDecoder for how {@code tClass} can be used.
```

Comment:
Since it's a `@see`, let's make the title a little more formal. Maybe add a header (`<h2>`) in the spec.